### DISCUSSION TITLE
Allow compilation without obs frontend

### DISCUSSION THREAD

**Discussion by RoboTricker** [iamscottxu/obs-rtspserver#138] [Ideas]
First of all thanks for this amazing plugin!

I'm interested in a version of this plugin that can be compiled without linking to any frontend libraries (neither obs-frontend nor Qt).
I would like to use the plugin in a standalone application which links to the core obs binaries but not to the obs frontend.

In case of the missing frontend I'd imagine that the plugin only creates the (rtsp-stream-)obs-output and leaves the responsibility for connecting a video source / encoder to the user-application.

Maybe a simple switch in the CMakeLists can allow building without the frontend. I'd be pleased to create a pull request with this change if you are interested.

**Comment by iamscottxu** [comment]
Thanks for the suggestion, you can create a pull request to do this, and add environment variables in CMakeLists to control whether to link frontend libraries.
